crates/modelardb_server/src/storage/types.rs

```diff
@@ -0,0 +1,478 @@
+        self.values.push_overwrite(new_value as u32);
+        self.last_value = new_value;
+    }
+
+    /// Return a reference to the metric's values for testing.
+    #[cfg(test)]
+    pub(super) fn values(&self) -> &HeapRb<u32> {
+        &self.values
+    }
+
+    /// Finish and reset the internal ring buffers and return the timestamps and values as Apache Arrow arrays.
+    pub(super) fn finish(&mut self) -> (TimestampArray, UInt32Array) {
+        let timestamps = TimestampArray::from_iter_values(self.timestamps.pop_iter());
+        let values = UInt32Array::from_iter_values(self.values.pop_iter());
+
+        (timestamps, values)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::common_test;
+
+    // Tests for MemoryPool.
+    #[test]
+    fn test_adjust_uncompressed_memory_increase() {
+        let memory_pool = create_memory_pool();
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES as isize
+        );
+
+        memory_pool.adjust_uncompressed_memory(common_test::COMPRESSED_SEGMENTS_SIZE as isize);
+
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            (common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES
+                + common_test::COMPRESSED_SEGMENTS_SIZE) as isize
+        );
+    }
+
+    #[test]
+    fn test_adjust_uncompressed_memory_decrease_above_zero() {
+        let memory_pool = create_memory_pool();
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES as isize
+        );
+
+        memory_pool.adjust_uncompressed_memory(-(common_test::COMPRESSED_SEGMENTS_SIZE as isize));
+
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            (common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES
+                - common_test::COMPRESSED_SEGMENTS_SIZE) as isize
+        );
+    }
+
+    #[test]
+    fn test_adjust_uncompressed_memory_decrease_below_zero() {
+        let memory_pool = create_memory_pool();
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES as isize
+        );
+
+        memory_pool.adjust_uncompressed_memory(
+            -2 * common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES as isize,
+        );
+
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            -(common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES as isize)
+        );
+    }
+
+    #[test]
+    fn test_try_reserve_uncompressed_memory_succeed() {
+        let memory_pool = create_memory_pool();
+        assert_eq!(
+            memory_pool.remaining_uncompressed_memory_in_bytes(),
+            common_test::UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES as isize
+        );
+
+        memory_pool
```

Comment:
To test that the function both makes the correct change and returns the correct boolean value, maybe add an `assert!` around all the calls to `memory_pool.try_reserve_uncompressed_memory` and `memory_pool.try_reserve_compressed_memory` in the tests.

Reply:
Good idea, fixed and added the [`must_use` attribute](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute).